src/nostr_service.cpp

```diff
@@ -182,104 +182,108 @@ tuple<vector<string>, vector<string>> NostrService::publishEvent(shared_ptr<Even
     return make_tuple(successfulRelays, failedRelays);
 };
 
-// TODO: Make this method return a promise.
 // TODO: Add a timeout to this method to prevent hanging while waiting for the relay.
-vector<shared_ptr<Event>> NostrService::queryRelays(shared_ptr<Filters> filters)
+std::future<vector<shared_ptr<Event>>> NostrService::queryRelays(shared_ptr<Filters> filters)
 {
-    if (filters->limit > 64 || filters->limit < 1)
-    {
-        PLOG_WARNING << "Filters limit must be between 1 and 64, inclusive.  Setting limit to 16.";
-        filters->limit = 16;
-    }
-
-    vector<shared_ptr<Event>> events;
-
-    string subscriptionId = this->generateSubscriptionId();
-    string request;
-
-    try
-    {
-        request = filters->serialize(subscriptionId);
-    }
-    catch (const invalid_argument& e)
-    {
-        PLOG_ERROR << "Failed to serialize filters - invalid object: " << e.what();
-        throw e;
-    }
-    catch (const json::exception& je)
+    return std::async(std::launch::async, [this, filters]() -> vector<shared_ptr<Event>>
```

Comment:
Same here.  Stylistically, I'm leaning away from explicitly declaring the `std::` namespace in implementation files when we don't have to.

```suggestion
    return async(launch::async, [this, filters]() -> vector<shared_ptr<Event>>
```

Reply:
Additionally, we may be better prepared for future work around timeouts by declaring a promise at the start of the function, and returning the future from that promise at the end.

That lets us fulfill the promise on timeout (eventually) instead of just by completing the query.